perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/RegistrarManagerImpl.java

```diff
@@ -3866,27 +3858,17 @@ private void storeApplicationAttributes(Application app) throws UserNotExistsExc
 	}
 
 	/**
-	 * Normalize input value from registration form for SSH keys and correctly merge them with the originalValue from User.
+	 * Normalize input value from registration form for array values and correctly merge them with the originalValue from User.
 	 *
 	 * @param originalValue Value from Users attribute currently stored in Perun
 	 * @param newValue Value provided by registration form
-	 * @return List of SSH keys after merge
+	 * @return List of array attribute values after merge
 	 */
-	private ArrayList<String> handleSSHKeysValue(ArrayList<String> originalValue, String newValue) {
-
-		// blank input means no change to original attribute
-		if (StringUtils.isBlank(newValue)) {
```

Comment:
You must keep this blank check and also use it in handling map values, since GUI send empty items as empty strings and without this later processing will fail, as it will try to store list with empty string as single/first value.

Reply:
Added this check to both methods.